packages/channels/botpress-channel-web/src/db.js

```diff
@@ -203,30 +204,37 @@ module.exports = knex => {
 
     const conversationIds = conversations.map(c => c.id)
 
+    const lastMessagesDate = (await knex('web_messages')
+      .whereIn('conversationId', conversationIds)
+      .groupBy('web_messages.conversationId')
+      .orderBy('sent_on', 'desc')
+      .select(knex.raw('max(sent_on) as date'))).map(mess => mess.date)
+
+    const lastMassegeInConversation = knex
+      .from('web_messages')
+      .select('*')
+      .whereIn('sent_on', lastMessagesDate)
+
     return knex
       .from(function() {
-        this.from('web_messages')
-          .whereIn('conversationId', conversationIds)
-          .groupBy('conversationId')
-          .select('conversationId', knex.raw('max(id) as msgid'))
-          .as('q1')
+        this.from('web_conversations')
+          .where({ userId })
+          .as('wc')
       })
-      .innerJoin('web_conversations', 'web_conversations.id', 'q1.conversationId')
-      .innerJoin('web_messages', 'web_messages.id', 'q1.msgid')
-      .orderBy('web_messages.sent_on', 'desc')
+      .leftJoin(lastMassegeInConversation.as('wm'), 'wm.conversationId', 'wc.id')
```

Comment:
As this query been tested on both SQLite and Postgres?

Reply:
@slvnperron , imlemented different queried for sqlite and postgres here